src/dfx/src/lib/models/canister.rs

```diff
@@ -338,30 +338,40 @@ fn wasm_opt_level_convert(opt_level: WasmOptLevel) -> OptLevel {
 }
 
 fn separate_candid(path: &Path) -> DfxResult<(String, String)> {
+    use candid::pretty::candid::{compile, pp_args};
+    use candid::types::internal::TypeInner;
+    use candid_parser::{
+        pretty_parse,
+        types::{Dec, IDLProg},
+    };
+    let did = dfx_core::fs::read_to_string(path)?;
+    let prog = pretty_parse::<IDLProg>(path.to_str().unwrap(), &did)?;
```

Comment:
Let's avoid unwrap if we can.  Something like this, or return an error.
```suggestion
    let prog = pretty_parse::<IDLProg>(&format!("{}", path.display()), &did)?;
```